### PR TITLE
Add input device manager for key grab

### PR DESCRIPTION
**Pull request by ideallee**
Add input device manager for handling "set/unset for key grab" when requested by  web device api

### CODE REVIEW

**Comment by WonyoungChoi** [comment]
Use "PascalCase" for member function names in each class.

File: runtime/browser/input_device_manager.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    KEY_EVENT_SUPPORT_ERROR_KEY_NOT_FOUND = -1,
+    KEY_EVENT_SUPPORT_ERROR_PRIVILEGE_LEVEL_VIOLATION = -2,
+    KEY_EVENT_SUPPORT_ERROR_IO_ERROR = -3,
+    KEY_EVENT_SUPPORT_ERROR_INVALID_EWK_VIEW = -4
+  } KeyEventSupportError;
+  static InputDeviceManager* GetInstance();
+  InputDeviceManager();
+  ~InputDeviceManager();
+  void initialize(Evas_Object* window);
+  bool registerKey(const std::string keyname);
```
Comment (WonyoungChoi) [inline runtime/browser/input_device_manager.h]: use const reference type

File: runtime/browser/input_device_manager.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  bool registerKey(const std::list<std::string>& keylist);
+  bool unregisterKey(const std::string keyname);
```
Comment (WonyoungChoi) [inline runtime/browser/input_device_manager.h]: use const reference type

File: runtime/browser/web_application.cc [hunk 3]
```diff
@@ -566,6 +568,92 @@ void WebApplication::OnReceivedWrtMessage(WebView* /*view*/,
...
     splash_screen_->HideSplashScreen(SplashScreen::HideReason::CUSTOM);
+  } else if (TYPE_IS("tizen://api/inputdevice/registerKey")) {
+    bool ret = false;
+    std::string key = msg_value ? msg_value : "";
+    ret = InputDeviceManager::GetInstance()->registerKey(key);
+    // Send response
+    Ewk_IPC_Wrt_Message_Data* ans = ewk_ipc_wrt_message_data_new();
+    ewk_ipc_wrt_message_data_type_set(ans, msg_type);
+    ewk_ipc_wrt_message_data_reference_id_set(ans, msg_id);
+    if (ret)
```
Comment (WonyoungChoi) [inline runtime/browser/web_application.cc]: put "brace" even the content has just one line.

File: runtime/browser/web_application.cc [hunk 4]
```diff
@@ -566,6 +568,92 @@ void WebApplication::OnReceivedWrtMessage(WebView* /*view*/,
...
+  } else if (TYPE_IS("tizen://api/inputdevice/unregisterKey")) {
+    bool ret = false;
+    std::string key = msg_value ? msg_value : "";
+    ret = InputDeviceManager::GetInstance()->unregisterKey(key);
+    // Send response
+    Ewk_IPC_Wrt_Message_Data* ans = ewk_ipc_wrt_message_data_new();
+    ewk_ipc_wrt_message_data_type_set(ans, msg_type);
+    ewk_ipc_wrt_message_data_reference_id_set(ans, msg_id);
+    if (ret)
+      ewk_ipc_wrt_message_data_value_set(ans, "success");
```
Comment (WonyoungChoi) [inline runtime/browser/web_application.cc:597]: put braces

File: runtime/browser/web_application.cc [hunk 5]
```diff
@@ -566,6 +568,92 @@ void WebApplication::OnReceivedWrtMessage(WebView* /*view*/,
...
+      ptr = strtok_r(NULL, ",", &saveptr);
+    }
+    free(buf);
+
+    ret = InputDeviceManager::GetInstance()->registerKey(keylist);
+
+    Ewk_IPC_Wrt_Message_Data* ans = ewk_ipc_wrt_message_data_new();
+    ewk_ipc_wrt_message_data_type_set(ans, msg_type);
+    ewk_ipc_wrt_message_data_reference_id_set(ans, msg_id);
+    if (ret)
```
Comment (WonyoungChoi) [inline runtime/browser/web_application.cc]: put braces

File: runtime/browser/web_application.cc [hunk 6]
```diff
@@ -566,6 +568,92 @@ void WebApplication::OnReceivedWrtMessage(WebView* /*view*/,
...
+      ewk_ipc_wrt_message_data_value_set(ans, "failed");
+    if (!ewk_ipc_wrt_message_send(ewk_context_, ans)) {
+      LOGGER(ERROR) << "Failed to send response";
+    }
+    ewk_ipc_wrt_message_data_del(ans);
+  } else if (TYPE_IS("tizen://api/inputdevice/registerKeyBatch")) {
+    std::list<std::string> keylist;
+    bool ret = false;
+
+    char *buf = strdup(msg_value ? msg_value : "");
```
Comment (WonyoungChoi) [inline runtime/browser/web_application.cc]: the pointer astrisk adjacent to the type. 
char\* buf , not char *buf

File: runtime/browser/web_application.cc [hunk 7]
```diff
@@ -566,6 +568,96 @@ void WebApplication::OnReceivedWrtMessage(WebView* /*view*/,
...
+      LOGGER(ERROR) << "Failed to send response";
+    }
+    ewk_ipc_wrt_message_data_del(ans);
+  } else if (TYPE_IS("tizen://api/inputdevice/unregisterKeyBatch")) {
+    std::list<std::string> keylist;
+    bool ret = false;
+
+    char* buf = strdup(msg_value ? msg_value : "");
+    char* saveptr = NULL;
+    const char* ptr = strtok_r(buf, ",", &saveptr);
```
Comment (myroot) [inline runtime/browser/web_application.cc]: Make High level C++ API to split string

see common/string_utils.h

**Pull request closed without merging**